clarity/contracts/vaults/arkadiko-stacker-payer-v3-8.clar

```diff
@@ -0,0 +1,129 @@
+;; Implementation of Stacker Payer
+;; which allows users to redeem xSTX for STX
+
+(define-constant ERR-NOT-AUTHORIZED u22401)
+(define-constant ERR-EMERGENCY-SHUTDOWN-ACTIVATED u221)
+(define-constant ERR-BURN-HEIGHT-NOT-REACHED u222)
+(define-constant ERR-WRONG-COLLATERAL-TOKEN u223)
+(define-constant ERR-VAULT-LIQUIDATED u224)
+(define-constant ERR-STILL-STACKING u225)
+
+(define-data-var stacker-payer-shutdown-activated bool false)
+
+(define-map unlocks { vault-id: uint } { unlocked-at-burn-height: uint })
+
+(define-read-only (get-vault-unlock (vault-id uint))
+  (default-to
+    {
+      unlocked-at-burn-height: u999999999999999
+    }
+    (map-get? unlocks { vault-id: vault-id })
+  )
+)
+
+(define-read-only (get-vault-combined-unlock (vault-id uint))
+  (let (
+    (old-unlock (contract-call? .arkadiko-stacker-payer-v3-7 get-vault-unlock vault-id))
+    (old-unlock-burn-height (get unlocked-at-burn-height old-unlock))
+    (new-unlock (get-vault-unlock vault-id))
+  )
+    (if (<= old-unlock-burn-height u829850)
```

Comment:
Q: Why this block?

Reply:
A: this is the end block of the next PoX cycle, meaning anyone who triggered an unlock on the `v3-7` contract will get this block set for them (and thus we need to respect that unlock for these users).

Users who unlock after will get a higher block, and thus will be using the `v3-8` `unlocks` map